src/ParametersHandler/YarpImplementation/src/YarpImplementation.cpp

```diff
@@ -6,33 +6,89 @@
  */
 
 #include <BipedalLocomotionControllers/ParametersHandler/YarpImplementation.h>
+#include <yarp/os/Bottle.h>
 
 #include <string>
+#include <cassert>
 
 using namespace BipedalLocomotionControllers::ParametersHandler;
 
 YarpImplementation::YarpImplementation(const yarp::os::Searchable& searchable)
 {
-    m_container.fromString(searchable.toString());
+    set(searchable);
 }
 
-std::unique_ptr<IParametersHandler<YarpImplementation>>
-YarpImplementation::getGroup(const std::string& name) const
+void YarpImplementation::set(const yarp::os::Searchable &searchable)
 {
-    auto& group = m_container.findGroup(name);
-    if (group.isNull())
-        return std::make_unique<YarpImplementation>();
+    clear();
 
-    return std::make_unique<YarpImplementation>(group);
+    yarp::os::Bottle bot;
+    bot.fromString(searchable.toString());
+
+    for (size_t i = 0; i < bot.size(); i++) { //all sublists are included in a new object
+        yarp::os::Value& bb = bot.get(i);
+
+        yarp::os::Bottle* sub = bb.asList();
+        if ((sub) && (sub->size() > 1)) {
+            std::string name = sub->get(0).toString();
+            yarp::os::Bottle* subSub = sub->get(1).asList();
+            if ((subSub) && (subSub->size() > 1))
+            {
+                m_lists.emplace(name, make_shared(*subSub));
+            }
+            else
+            {
+                m_container.add(bb);
+            }
+
+        }
+        else
+        {
+            m_container.add(bb);
+        }
+    }
+}
+
+YarpImplementation::weak_ptr YarpImplementation::getGroup(const std::string& name) const
+{
+    if (m_lists.find(name) != m_lists.end())
+    {
+        return m_lists.at(name);
+    }
+
+    return YarpImplementation::make_shared();
```

Comment:
Correct me if I'm mistaken. if the group does not exist. A new `shared_ptr` should be added to `m_list`
```cpp
 m_lists.emplace(name, make_shared(*subSub));
```
Indeed if the new group is not added in the list the `lock()` method of the weak pointer should fail 


```cpp
int main()
{
    ....
    YarpImplementation::weak_ptr  weakPtr = handler->getGroup("foo"); // <---- foo is not a name of
                                                                      //       any group 

    if(auto ptr = weakPtr.lock())
        std::cout << ptr->toString() << std::endl;
     else                                      // <----- according to the implemetation  of the 
                                               //        YarpImplementation::getGroup the pointer
                                               //        should be expired 
        std::cout << "pointer expired" << std::endl;
    ....
}
```

Reply:
> if the group does not exist. A new shared_ptr should be added to m_list

I cannot, the method is ``const``.

> Indeed if the new group is not added in the list the lock() method of the weak pointer should fail

This is exactly what I want. Thus we have a way to understand if the group actually exists or not. To create a new group you can use the ``setGroup`` method.

Reply:
I realized this was not clear from the documentation. I changed it with https://github.com/dic-iit/bipedal-locomotion-controllers/pull/18/commits/b2197cdec17cc3c5454bfbb93961c5951f837111